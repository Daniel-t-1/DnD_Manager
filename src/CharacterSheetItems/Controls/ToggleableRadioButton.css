.Radiobutton-On {
  border-radius: 50%;
  height: 20px;
  width: 20px;
  border-style: solid;
  border-width: 2px;
  border-color: black;
  color: white;
  background-color: white;
  padding: 4px;
}

.Radiobutton-Off {
  border-radius: 50%;
  height: 20px;
  width: 20px;
  border-style: solid;
  border-color: black;
  color: black;
  background-color: black;
  background-clip: content-box; /* support: IE9+ */
  padding: 4px;
}

.Radiobutton-Off:hover,
.Radiobutton-On:hover {
  cursor: pointer;
  height: 21px;
  width: 21px;
}
