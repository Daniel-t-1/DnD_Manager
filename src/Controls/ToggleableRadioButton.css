.radiobutton--off {
  border-radius: 50%;
  height: 20px;
  width: 20px;
  border-style: solid;
  border-width: 2px;
  border-color: black;
  color: white;
  background-color: white;
  padding: 4px;
  transition-property: width, height;
  transition-duration: 0.3s;
}

.radiobutton--on {
  border-radius: 50%;
  height: 20px;
  width: 20px;
  border-style: solid;
  border-color: black;
  color: black;
  background-color: black;
  background-clip: content-box; /* support: IE9+ */
  padding: 4px;
  transition-property: width, height;
  transition-duration: 0.3s;
}

.radiobutton--off:hover,
.radiobutton--on:hover {
  cursor: pointer;
  height: 21px;
  width: 21px;
  transition-property: width, height;
  transition-duration: 0s;
}

.radiobutton-small--on:hover,
.radiobutton-small--off:hover {
  cursor: pointer;
  transition-property: width, height;
  transition-duration: 0s;
  height: 17px;
  width: 17px;
}

.radiobutton-small--off {
  height: 15px;
  width: 15px;
  border-style: solid;
  border-width: 2px;
  border-color: black;
  color: white;
  background-color: white;
  padding: 4px;
  transition-property: width, height;
  transition-duration: 0.3s;
  border-radius: 50%;
}

.radiobutton-small--on {
  height: 15px;
  width: 15px;
  padding: 2px;
  border-radius: 50%;
  border-style: solid;
  border-color: black;
  color: black;
  background-color: black;
  background-clip: content-box; /* support: IE9+ */
  transition-property: width, height;
  transition-duration: 0.3s;
}
