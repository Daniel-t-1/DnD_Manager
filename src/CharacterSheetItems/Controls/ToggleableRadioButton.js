import React from "react";
import "./ToggleableRadioButton.css";

export function ToggleableRadioButton(props) {
  const toggle = () => {
    props.updateChecked(!props.enabled);
  };

  return (
    <button
      type="button"
      className={props.enabled ? "Radiobutton-On" : "Radiobutton-Off"}
      onClick={() => toggle()}
    ></button>
  );
}

export default ToggleableRadioButton;
