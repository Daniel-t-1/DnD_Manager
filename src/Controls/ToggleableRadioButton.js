import React from "react";
import "./ToggleableRadioButton.css";

export function ToggleableRadioButton({ enabled, id, updateChecked, size }) {
  const toggle = () => {
    updateChecked(!enabled, id);
  };

  const sizestring = size === "small" ? "radiobutton-small" : "radiobutton";

  return (
    <button
      type="button"
      data-testid="ToggleableRadioButton"
      className={enabled ? sizestring + "--on" : sizestring + "--off"}
      onClick={() => toggle()}
    ></button>
  );
}

export default ToggleableRadioButton;
