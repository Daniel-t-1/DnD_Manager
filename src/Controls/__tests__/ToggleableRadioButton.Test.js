import { render, screen } from "@testing-library/react";
import ToggleableRadioButton from "../ToggleableRadioButton";

test("Renders without crashing", () => {
  render(<ToggleableRadioButton />);
});

//test to see if the button is rendered
test("Render returns Button", () => {
  render(<ToggleableRadioButton />);
  const linkElement = screen.getByTestId("ToggleableRadioButton");
  expect(linkElement).toBeInTheDocument();
});

//Test button calls updateChecked
test("Button calls updateChecked", () => {
  const updateChecked = jest.fn();
  render(<ToggleableRadioButton updateChecked={updateChecked} />);
  const button = screen.getByTestId("ToggleableRadioButton");
  button.click();
  expect(updateChecked).toHaveBeenCalled();
});

//snapshot testing
test("Snapshot testing button On", () => {
  const { asFragment } = render(<ToggleableRadioButton enabled={true} />);
  expect(asFragment()).toMatchSnapshot();
});

test("Snapshot testing button Off", () => {
  const { asFragment } = render(<ToggleableRadioButton enabled={false} />);
  expect(asFragment()).toMatchSnapshot();
});

test("Snapshot testing small button Off", () => {
  const { asFragment } = render(
    <ToggleableRadioButton enabled={false} size="small" />
  );
  expect(asFragment()).toMatchSnapshot();
});
test("Snapshot testing Small button On", () => {
  const { asFragment } = render(
    <ToggleableRadioButton enabled={true} size="small" />
  );
  expect(asFragment()).toMatchSnapshot();
});
