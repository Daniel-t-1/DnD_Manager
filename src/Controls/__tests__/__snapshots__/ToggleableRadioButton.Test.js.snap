// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Small button On 1`] = `
<DocumentFragment>
  <button
    class="Radiobutton-Small-On"
    data-testid="ToggleableRadioButton"
    type="button"
  />
</DocumentFragment>
`;

exports[`Snapshot testing button Off 1`] = `
<DocumentFragment>
  <button
    class="Radiobutton-Off"
    data-testid="ToggleableRadioButton"
    type="button"
  />
</DocumentFragment>
`;

exports[`Snapshot testing button On 1`] = `
<DocumentFragment>
  <button
    class="Radiobutton-On"
    data-testid="ToggleableRadioButton"
    type="button"
  />
</DocumentFragment>
`;

exports[`Snapshot testing small button Off 1`] = `
<DocumentFragment>
  <button
    class="Radiobutton-Small-Off"
    data-testid="ToggleableRadioButton"
    type="button"
  />
</DocumentFragment>
`;
